RecoTracker/TkNavigation: fixes for these clang warnings -Woverloaded-virtual

gartung opened the pull request:
In file included from /build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TkNavigation/plugins/SimpleNavigationSchool.cc:5:
  /build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TkNavigation/plugins/SimpleBarrelNavigableLayer.h:34:3: warning: 'SimpleBarrelNavigableLayer::compatibleLayers' hides overloaded virtual function [-Woverloaded-virtual]
   compatibleLayers( NavigationDirection direction) const override;
  ^
/build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TkNavigation/plugins/SimpleNavigableLayer.h:36:35: note: hidden overloaded virtual function 'SimpleNavigableLayer::compatibleLayers' declared here: different number of parameters (3 vs 1)
  std::vector< const DetLayer * > compatibleLayers (const FreeTrajectoryState &fts,
                                  ^
In file included from /build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TkNavigation/plugins/SimpleNavigationSchool.cc:5:
  /build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TkNavigation/plugins/SimpleBarrelNavigableLayer.h:37:3: warning: 'SimpleBarrelNavigableLayer::compatibleLayers' hides overloaded virtual function [-Woverloaded-virtual]
   compatibleLayers( const FreeTrajectoryState& fts,
  ^
/build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TkNavigation/plugins/SimpleNavigableLayer.h:36:35: note: hidden overloaded virtual function 'SimpleNavigableLayer::compatibleLayers' declared here: different number of parameters (3 vs 2)
  std::vector< const DetLayer * > compatibleLayers (const FreeTrajectoryState &fts,
                                  ^
In file included from /build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TkNavigation/plugins/SimpleNavigationSchool.cc:6:
  /build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TkNavigation/plugins/SimpleForwardNavigableLayer.h:29:3: warning: 'SimpleForwardNavigableLayer::compatibleLayers' hides overloaded virtual function [-Woverloaded-virtual]
   compatibleLayers( NavigationDirection direction) const override;
  ^
/build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TkNavigation/plugins/SimpleNavigableLayer.h:36:35: note: hidden overloaded virtual function 'SimpleNavigableLayer::compatibleLayers' declared here: different number of parameters (3 vs 1)
  std::vector< const DetLayer * > compatibleLayers (const FreeTrajectoryState &fts,
                                  ^
In file included from /build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TkNavigation/plugins/SimpleNavigationSchool.cc:6:
  /build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TkNavigation/plugins/SimpleForwardNavigableLayer.h:32:3: warning: 'SimpleForwardNavigableLayer::compatibleLayers' hides overloaded virtual function [-Woverloaded-virtual]
   compatibleLayers( const FreeTrajectoryState& fts,
  ^
/build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TkNavigation/plugins/SimpleNavigableLayer.h:36:35: note: hidden overloaded virtual function 'SimpleNavigableLayer::compatibleLayers' declared here: different number of parameters (3 vs 2)
  std::vector< const DetLayer * > compatibleLayers (const FreeTrajectoryState &fts,
                                  ^

gartung committed:
RecoTracker/TkNavigation: fixes for these clanf warnings -Woverloaded-virtual

In file included from /build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TkNavigation/plugins/SimpleNavigationSchool.cc:5:
  /build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TkNavigation/plugins/SimpleBarrelNavigableLayer.h:34:3: warning: 'SimpleBarrelNavigableLayer::compatibleLayers' hides overloaded virtual function [-Woverloaded-virtual]
   compatibleLayers( NavigationDirection direction) const override;
  ^
/build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TkNavigation/plugins/SimpleNavigableLayer.h:36:35: note: hidden overloaded virtual function 'SimpleNavigableLayer::compatibleLayers' declared here: different number of parameters (3 vs 1)
  std::vector< const DetLayer * > compatibleLayers (const FreeTrajectoryState &fts,
                                  ^
In file included from /build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TkNavigation/plugins/SimpleNavigationSchool.cc:5:
  /build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TkNavigation/plugins/SimpleBarrelNavigableLayer.h:37:3: warning: 'SimpleBarrelNavigableLayer::compatibleLayers' hides overloaded virtual function [-Woverloaded-virtual]
   compatibleLayers( const FreeTrajectoryState& fts,
  ^
/build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TkNavigation/plugins/SimpleNavigableLayer.h:36:35: note: hidden overloaded virtual function 'SimpleNavigableLayer::compatibleLayers' declared here: different number of parameters (3 vs 2)
  std::vector< const DetLayer * > compatibleLayers (const FreeTrajectoryState &fts,
                                  ^
In file included from /build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TkNavigation/plugins/SimpleNavigationSchool.cc:6:
  /build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TkNavigation/plugins/SimpleForwardNavigableLayer.h:29:3: warning: 'SimpleForwardNavigableLayer::compatibleLayers' hides overloaded virtual function [-Woverloaded-virtual]
   compatibleLayers( NavigationDirection direction) const override;
  ^
/build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TkNavigation/plugins/SimpleNavigableLayer.h:36:35: note: hidden overloaded virtual function 'SimpleNavigableLayer::compatibleLayers' declared here: different number of parameters (3 vs 1)
  std::vector< const DetLayer * > compatibleLayers (const FreeTrajectoryState &fts,
                                  ^
In file included from /build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TkNavigation/plugins/SimpleNavigationSchool.cc:6:
  /build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TkNavigation/plugins/SimpleForwardNavigableLayer.h:32:3: warning: 'SimpleForwardNavigableLayer::compatibleLayers' hides overloaded virtual function [-Woverloaded-virtual]
   compatibleLayers( const FreeTrajectoryState& fts,
  ^
/build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TkNavigation/plugins/SimpleNavigableLayer.h:36:35: note: hidden overloaded virtual function 'SimpleNavigableLayer::compatibleLayers' declared here: different number of parameters (3 vs 2)
  std::vector< const DetLayer * > compatibleLayers (const FreeTrajectoryState &fts,
                                  ^

gartung committed:
RecoTracker/TrackProducer: fix for clang warning -Woverloaded-virtual

In file included from /build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TrackProducer/plugins/SealModules.cc:3:
  /build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TrackProducer/plugins/DAFTrackProducer.h:29:8: warning: 'DAFTrackProducer::getFromEvt' hides overloaded virtual functions [-Woverloaded-virtual]
   void getFromEvt(edm::Event&, edm::Handle<TrajTrackAssociationCollection>&, reco::BeamSpot&);
       ^
/build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TrackProducer/interface/TrackProducerBase.h:66:16: note: hidden overloaded virtual function 'TrackProducerBase<reco::Track>::getFromEvt' declared here: type mismatch at 2nd parameter ('edm::Handle<TrackCandidateCollection> &' (aka 'Handle<vector<TrackCandidate> > &') vs 'edm::Handle<TrajTrackAssociationCollection> &' (aka 'Handle<AssociationMap<OneToOne<vector<Trajectory>, vector<reco::Track>, unsigned short> > > &'))
  virtual void getFromEvt(edm::Event&, edm::Handle<TrackCandidateCollection>&, reco::BeamSpot&);
               ^
/build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c6796210c53b02f1870cdf07f5d562c4/opt/cmssw/slc6_amd64_gcc630/cms/cmssw/CMSSW_10_1_CLANG_X_2018-03-26-2300/src/RecoTracker/TrackProducer/interface/TrackProducerBase.h:68:16: note: hidden overloaded virtual function 'TrackProducerBase<reco::Track>::getFromEvt' declared here: type mismatch at 2nd parameter ('edm::Handle<TrackView> &' (aka 'Handle<View<reco::Track> > &') vs 'edm::Handle<TrajTrackAssociationCollection> &' (aka 'Handle<AssociationMap<OneToOne<vector<Trajectory>, vector<reco::Track>, unsigned short> > > &'))
  virtual void getFromEvt(edm::Event&, edm::Handle<TrackView>&, reco::BeamSpot&);
               ^

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-22765/4137

cmsbuild commented:
A new Pull Request was created by  @gartung (Patrick Gartung) for master.

It involves the following packages:

RecoTracker/TkNavigation


@perrotta, @cmsbuild, @slava77 can you please review it and eventually sign? Thanks.
@ghellwig, @makortel, @felicepantaleo, @GiacomoSguazzoni, @rovere, @VinInn, @mschrode, @gpetruc, @ebrondol, @dgulhan this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

perrotta commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 96433167fd2a0157ae62b00a6b0e70203bdc3f12
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22765/27150/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22765/27150/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 29
 - DQMHistoTests: Total histograms compared: 2501580
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2501403
 - DQMHistoTests: Total skipped: 176
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.97000000019 KiB( 23 files compared)
 - Checked 118 log files, 9 edm output root files, 29 DQM output files

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/22765	9643316
- technical 
- jenkins tests pass

I don't fully follow the logic of the warning.
The 3-parameter method from SimpleNavigableLayer is declared final there. So, it would naively make sense not to complain about it being hidden.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)

gartung committed:
RecoTracker/TkNavigation: fix clang warning: -Woverloaded-virtual

src/RecoTracker/TkNavigation/plugins/SimpleBarrelNavigableLayer.h:34:3: warning: 'SimpleBarrelNavigableLayer::compatibleLayers' hides overloaded virtual function [-Woverloaded-virtual]
   compatibleLayers( NavigationDirection direction) const override;
  ^
src/RecoTracker/TkNavigation/plugins/SimpleNavigableLayer.h:36:35: note: hidden overloaded virtual function 'SimpleNavigableLayer::compatibleLayers' declared here: different number of parameters (3 vs 1)
  std::vector< const DetLayer * > compatibleLayers (const FreeTrajectoryState &fts,
                                  ^

src/RecoTracker/TkNavigation/plugins/SimpleBarrelNavigableLayer.h:37:3: warning: 'SimpleBarrelNavigableLayer::compatibleLayers' hides overloaded virtual function [-Woverloaded-virtual]
   compatibleLayers( const FreeTrajectoryState& fts,
  ^
src/RecoTracker/TkNavigation/plugins/SimpleNavigableLayer.h:36:35: note: hidden overloaded virtual function 'SimpleNavigableLayer::compatibleLayers' declared here: different number of parameters (3 vs 2)
  std::vector< const DetLayer * > compatibleLayers (const FreeTrajectoryState &fts,
                                  ^

src/RecoTracker/TkNavigation/plugins/SimpleForwardNavigableLayer.h:29:3: warning: 'SimpleForwardNavigableLayer::compatibleLayers' hides overloaded virtual function [-Woverloaded-virtual]
   compatibleLayers( NavigationDirection direction) const override;
  ^
src/RecoTracker/TkNavigation/plugins/SimpleNavigableLayer.h:36:35: note: hidden overloaded virtual function 'SimpleNavigableLayer::compatibleLayers' declared here: different number of parameters (3 vs 1)
  std::vector< const DetLayer * > compatibleLayers (const FreeTrajectoryState &fts,
                                  ^

src/RecoTracker/TkNavigation/plugins/SimpleForwardNavigableLayer.h:32:3: warning: 'SimpleForwardNavigableLayer::compatibleLayers' hides overloaded virtual function [-Woverloaded-virtual]
   compatibleLayers( const FreeTrajectoryState& fts,
  ^
src/RecoTracker/TkNavigation/plugins/SimpleNavigableLayer.h:36:35: note: hidden overloaded virtual function 'SimpleNavigableLayer::compatibleLayers' declared here: different number of parameters (3 vs 2)
  std::vector< const DetLayer * > compatibleLayers (const FreeTrajectoryState &fts,
                                  ^

src/RecoTracker/TkNavigation/plugins/CosmicNavigationSchool.cc:47:8: warning: 'CosmicNavigationSchool::establishInverseRelations' hides overloaded virtual function [-Woverloaded-virtual]
   void establishInverseRelations( SymmetricLayerFinder& symFinder );
       ^
src/RecoTracker/TkNavigation/plugins/SimpleNavigationSchool.h:73:16: note: hidden overloaded virtual function 'SimpleNavigationSchool::establishInverseRelations' declared here: different number of parameters (0 vs 1)
  virtual void establishInverseRelations();
               ^

gartung committed:
Merge branch 'RecoTracker-TkNavigation-clang-warn-fix' into RecoTracker-TkNavigation-fix-clang-warn

gartung committed:
Merge branch 'RecoTracker-TkNavigation-fix-clang-warn' of https://github.com/gartung/cmssw into RecoTracker-TkNavigation-fix-clang-warn

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-22765/4161

Code check has found code style and quality issues which could be resolved by applying a patch in https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-22765/4161/git-diff.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-22765/4161/git-diff.patch | patch -p1`

You can run `scram build code-checks` to apply code checks directly

gartung committed:
code-checks

cmsbuild commented:
The code-checks are being triggered in jenkins.

gartung committed:
added by accident

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-22765/4162

cmsbuild commented:
Pull request #22765 was updated. @perrotta, @cmsbuild, @slava77 can you please check and sign again.

gartung commented:
This is FUBAR. Resubmitting with cherry-picked commits
